zqd/handlers.go

```diff
@@ -43,6 +43,16 @@ func handleSearch(c *Core, w http.ResponseWriter, r *http.Request) {
 	}
 	defer cancel()
 
+	// XXX This always returns bad request but should return status codes
+	// that reflect the nature of the returned error.
+	w.Header().Set("Content-Type", "application/ndjson")
```

Comment:
Shouldn't this happen only if http.Error isn't called?

Reply:
good catch, thanks!